pipeline_tools/get_analysis_metadata.py

```diff
@@ -36,17 +37,29 @@ def get_auth(credentials_file=None):
 
 
 @retry(reraise=True, wait=wait_exponential(multiplier=1, max=10), stop=stop_after_delay(20))
-def get_metadata(runtime_environment, workflow_id):
+def get_metadata(runtime_environment, workflow_id, use_caas, caas_key_file=None):
     """
     Get metadata for analysis workflow from Cromwell and write it to a JSON file. Retry the request with exponentially
     increasing wait times if there is an error.
     :param str runtime_environment: The cromwell environment the workflow was run in.
     :param str workflow_id: The analysis workflow id.
+    :param bool use_caas: Whether or not to use Cromwell-as-a-Service.
+    :param str caas_key_file: Path to CAAS service account JSON key file.
     :return:
     """
     print('Getting metadata for workflow {}'.format(workflow_id))
-    url = 'https://cromwell.mint-{}.broadinstitute.org/api/workflows/v1/{}/metadata?expandSubWorkflows=true'.format(runtime_environment, workflow_id)
-    response = requests.get(url, auth=get_auth())
+    if use_caas:
+        cromwell_url = 'https://cromwell.caas-dev.broadinstitute.org/api/workflows/v1'
+        json_credentials = caas_key_file or "/cromwell-metadata/caas_key.json"
+        headers = cromwell_tools.generate_auth_header_from_key_file(json_credentials)
+        auth = None
+    else:
+        cromwell_url = 'https://cromwell.mint-{}.broadinstitute.org/api/workflows/v1'.format(runtime_environment)
+        headers = None
+        auth = get_auth()
+
+    url = '{}/{}/metadata?expandSubWorkflows=true'.format(cromwell_url, workflow_id)
```

Comment:
Just FYI: recently I ran into an error when I was trying to use "expandSubWorkflows": https://github.com/broadinstitute/cromwell/issues/3209 This error should be fixed in newer Cromwell versions, but not in our current deployed Cromwell instances.